include/solvers/navier_stokes_scratch_data.h

```diff
@@ -839,14 +921,167 @@ class NavierStokesScratchData : public PhysicsScratchDataBase
+  void
+  reinit_particle_fluid_interactions(
+    const typename DoFHandler<dim>::active_cell_iterator &velocity_cell,
+    const typename DoFHandler<dim>::active_cell_iterator &void_fraction_cell,
+    const VectorType & /*current_solution*/,
+    const VectorType                      &previous_solution,
+    const VectorType                      &void_fraction_solution,
+    const Particles::ParticleHandler<dim> &particle_handler)
+  {
+    pic = particle_handler.particles_in_cell(this->fe_values.get_cell());
+
+    // Calculate the particles properties
+    double total_particle_volume = 0;
+    total_particle_volume        = extract_particle_properties();
+
+    calculate_cell_void_fraction(total_particle_volume);
+
+    if (number_of_particles == 0)
+      return;
+
+    Quadrature<dim> q_particles = gather_particles_reference_location();
+    calculate_fluid_fields_at_particle_location(q_particles,
+                                                velocity_cell,
+                                                previous_solution);
+
+    // Create a quadrature for the void fraction that is based on the particle
+    // reference location
+    if (this->interpolated_void_fraction)
+      {
+        calculate_void_fraction_at_particle_location(q_particles,
+                                                     void_fraction_cell,
+                                                     void_fraction_solution);
+      }
+    calculate_fluid_properties_at_particle_location();
+    calculate_force_parameters_at_particle_location();
+  }
+
+  // Version of the function when VOF is used with CFD-DEM
+  /** @brief Calculates the variables needed to compute the particle fluid interactions
+   * in the vans equations.This version of the function is used when VOF is used
+   * with CFD-DEM.
+   *
+   * @param previous_solution The solution at the previous time step for the fluid's velocity
+   * and pressure
+   *
+   * @param void_fraction_solution The void fraction value calculated with one of the methods of
+   * the VoidFractionBase class.
+   *
+   * @param particle_handler The particle handler object that stores and manages the
+   *  particles in the simulations
+   *
+   * @param dof_handler The DoFHandler object for the Navier-Stokes problem
+   *
+   * @param void_fraction_dof_handler The DoFHandler object for the void fraction calculation
+   *
+   * @param dof_handler_vof The DoFHandler object for the VOF equation
+   *
+   * @param current_filtered_solution The present value of the phase fraction at the dofs of the
+   * cell
+   */
+  template <typename VectorType>
+  void
+  reinit_particle_fluid_interactions(
+    const typename DoFHandler<dim>::active_cell_iterator &velocity_cell,
+    const typename DoFHandler<dim>::active_cell_iterator &void_fraction_cell,
+    const typename DoFHandler<dim>::active_cell_iterator &phase_cell,
+    const VectorType & /*current_solution*/,
+    const VectorType                      &previous_solution,
```

Comment:
Maybe a good idea to call the solution fluid_dynamics_solution and previous solutions here to be more clear.